### ISSUE TITLE
alicloud_nat_gateway specification keeps updating in-place

### ISSUE THREAD

**Issue by fabien-duolingo** [aliyun/terraform-provider-alicloud#3372]
Hi there,

On the `alicloud_nat_gateway` resource, Terraform wants to update the `specification` at each run. 

### Terraform Version
```
Terraform v0.12.29
+ provider.alicloud v1.119.0
```
Not running the last Terraform version but I believe this is an issue with the API/provider. 
Running the last alicloud provider version (`1.119.0`)

### Affected Resource(s)
- alicloud_nat_gateway

### Terraform Configuration Files
```hcl
resource "alicloud_nat_gateway" "ngw-sz" {
  vpc_id        = alicloud_vpc.main.id
  nat_type      = "Enhanced"
  specification = "Small"
  vswitch_id    = alicloud_vswitch.vsw-szx-e.id
}
```

### Expected Behavior
After apply, the next `terraform plan` shows "No changes. Infrastructure is up-to-date."

### Actual Behavior
Terraform wants to update in-place the `alicloud_nat_gateway` resource : 
```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # alicloud_nat_gateway.ngw-sz will be updated in-place
  ~ resource "alicloud_nat_gateway" "ngw-sz" {
        forward_table_ids    = "ftb-abdc1234"
        id                   = "ngw-abdc1234"
        instance_charge_type = "PostPaid"
        nat_type             = "Enhanced"
        snat_table_ids       = "stb-abdc1234"
      + specification        = "Small"
        vpc_id               = "vpc-abdc1234"
        vswitch_id           = "vsw-abdc1234"
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Create a new enhanced NAT gateway from the console. 
2. Import with `terraform import alicloud_nat_gateway.ngw-sz ngw-abcd1234`
3. Run `terraform apply`, approve with `yes`
4. Run `terraform apply` again

### Important Factoids
- The issue does not happen if the gateway was upgraded from standard to enhanced from the console. Only if a new (enhanced) gateway is created from the console and then imported. 
- The *Specs* attribute shows as `-` on the UI instead of `Small`, `Middle` or `Large` as specified on the documentation. This might be what causes the issue.

**Comment by AllenLu55** [comment]
The cause of  this problem is PayByLcu will not return the specification unless PayBySpec.
So please remove the  "+ specification        = "Small" "  and test again.

**Comment by fabien-duolingo** [comment]
> The cause of this problem is PayByLcu will not return the specification unless PayBySpec.
> So please remove the "+ specification = "Small" " and test again.

I did try that, it does not solve the problem. The output is the same : 
```
------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # alicloud_nat_gateway.ngw-sz will be updated in-place
  ~ resource "alicloud_nat_gateway" "ngw-sz" {
        forward_table_ids    = "ftb-abcd1234"
        id                   = "ngw-abcd1234"
        instance_charge_type = "PostPaid"
        nat_type             = "Enhanced"
        snat_table_ids       = "stb-abcd1234"
      + specification        = "Small"
        vpc_id               = "vpc-abcd1234"
        vswitch_id           = "vsw-abcd1234"
    }

Plan: 0 to add, 1 to change, 0 to destroy.

------------------------------------------------------------------------
```

And the behavior is also the same as previously, if I apply I still have the same output during the next run.

**Comment by fabien-duolingo** [comment]
I ended up deleting and recreating the NAT Gateway directly from terraform (instead of creating from the console and importing afterwards). Like this I finally get the expected behavior. Terraform doesn't want to update-in-place the resource anymore on subsequent runs.

**Comment by xiaozhu36** [comment]
@fabien-duolingo This issue has been fixed by the latest version 1.121.0. Please have a check.

**Comment by xiaozhu36** [comment]
@fabien-duolingo I will close this issue and let me know or reopen it if you have more issue.
